src/R/Components/Impl/Plots/Implementation/Commands/CopyPlotAsBitmapCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Windows;
+using Microsoft.Common.Core;
+using Microsoft.Common.Wpf.Imaging;
+using Microsoft.R.Components.Controller;
+using Microsoft.R.Components.InteractiveWorkflow;
+
+namespace Microsoft.R.Components.Plots.Implementation.Commands {
+    internal sealed class CopyPlotAsBitmapCommand : PlotCommand, IAsyncCommand {
+        public CopyPlotAsBitmapCommand(IRInteractiveWorkflow interactiveWorkflow) : base(interactiveWorkflow) {
+        }
+
+        public CommandStatus Status {
+            get {
+                if (HasCurrentPlot && !IsInLocatorMode) {
+                    return CommandStatus.SupportedAndEnabled;
+                }
+
+                return CommandStatus.Supported;
+            }
+        }
+
+        public async Task<CommandResult> InvokeAsync() {
+            string filePath = Path.GetTempFileName();
+            try {
+                await InteractiveWorkflow.Plots.ExportToBitmapAsync("bmp", filePath);
+
+                InteractiveWorkflow.Shell.DispatchOnUIThread(() => {
+                    try {
+                        var image = BitmapImageFactory.Load(filePath);
+                        Clipboard.SetImage(image);
+                    } catch (Exception e) when (!e.IsCriticalException()) {
+                        MessageBox.Show(string.Format(Resources.Plots_CopyToClipboardError, e.Message));
```

Comment:
Should go via Shell.ShowErrorMessage or it may or may not appear or may appear behind VS